kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -163,6 +193,53 @@ public Transaction build(Engine engine) {
       }
     }
 
+    /* --------------- Handle Domain Metadata ------------ */
+    // Enable the feature if necessary
+    if (!TableFeatures.isDomainMetadataSupported(protocol)) {
+      if (!domainMetadatasAdded.isEmpty()) {
+        // This txn is setting a domain metadata, enable the feature in the protocol
+        logger.info(
+            "Automatically enabling writer feature: {}", DOMAIN_METADATA_W_FEATURE.featureName());
+        protocol =
+            protocol.withNewWriterFeatures(
+                Collections.singleton(DOMAIN_METADATA_W_FEATURE.featureName()));
+        shouldUpdateProtocol = true;
+      }
+      if (!domainMetadatasRemoved.isEmpty()) {
+        // Do nothing here, we will throw an exception when the domain does not exist in the
+        // current snapshot
+      }
+    }
+    // Generate the tombstones for the removed domain metadatas
+    Map<String, DomainMetadata> snapshotDomainMetadataMap = snapshot.getDomainMetadataMap();
+    for (String domainName : domainMetadatasRemoved) {
+      if (snapshotDomainMetadataMap.containsKey(domainName)) {
```

Comment:
if the names are case sensitive, this is a problem

Reply:
sorry how is this a problem if they are case sensitive (which they are)?